ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java

```diff
@@ -0,0 +1,1240 @@
+    return new Query((QueryBody) process(node.getQueryBody(), context),
+        node.getLimit());
+  }
+
+  protected Node visitExplain(Explain node, Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitShowCatalogs(ShowCatalogs node, Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitShowColumns(ShowColumns node, Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitShowPartitions(ShowPartitions node, Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitShowCreate(ShowCreate node, Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitShowFunctions(ShowFunctions node, Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitSetSession(SetSession node, Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitGenericLiteral(GenericLiteral node, Object context) {
+    return visitLiteral(node, context);
+  }
+
+  protected Node visitTimeLiteral(TimeLiteral node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new TimeLiteral(node.getLocation().get(), node.getValue());
+    } else {
+      return new TimeLiteral(node.getValue());
+    }
+
+  }
+
+  protected Node visitExplainOption(ExplainOption node, Object context) {
+    return visitNode(node, context);
+  }
+
+  protected Node visitWithQuery(WithQuery node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new WithQuery(node.getLocation().get(),
+          node.getName(),
+          (Query) process(node.getQuery(), context),
+          node.getColumnNames());
+    } else {
+      return new WithQuery(node.getName(),
+          (Query) process(node.getQuery(), context),
+          node.getColumnNames());
+    }
+
+  }
+
+  protected Node visitSelect(Select node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new Select(node.getLocation().get(),
+          node.isDistinct(),
+          node.getSelectItems()
+              .stream()
+              .map(selectItem -> (SelectItem) process(selectItem, context))
+              .collect(Collectors.toList())
+      );
+    } else {
+      return new Select(node.isDistinct(),
+          node.getSelectItems()
+              .stream()
+              .map(selectItem -> (SelectItem) process(selectItem, context))
+              .collect(Collectors.toList())
+      );
+    }
+
+  }
+
+  protected Node visitRelation(Relation node, Object context) {
+    return visitNode(node, context);
+  }
+
+  protected Node visitQueryBody(QueryBody node, Object context) {
+    return visitRelation(node, context);
+  }
+
+  protected Node visitQuerySpecification(QuerySpecification node, Object context) {
+    Optional<WindowExpression> windowExpression = node.getWindowExpression().isPresent()
```

Comment:
This can be simplified to:

```
Optional<WindowExpression> windowExpression = node.getWindowExpression()
   .flatMap((WindowExpression) process(node.getWindowExpression().get(), context));
```

Others elsewhere the same.

Reply:
We won't be able to use this since process(node.getWindowExpression().get(), context does not return optional and we need to check node.getWindowExpression() before calling get().

Reply:
Sorry, bad code example:

```
Optional<WindowExpression> windowExpression = node.getWindowExpression()
   .map(expression -> (WindowExpression) process(expression, context));
``